test/ATokenCheckpointed.t.sol

```diff
@@ -0,0 +1,1974 @@
+
+    // Advance the clock.
+    uint256 _blocksJumpedSecondTime = 100;
+    vm.roll(block.number + _blocksJumpedSecondTime);
+    vm.warp(block.timestamp + 100 days);
+
+    // Rebasing should not affect the raw balance.
+    assertEq(aToken.exposedRawBalanceOf(_who), _rawBalances[1]);
+    assertGt(_rawBalances[1], _rawBalances[0], "raw balance did not increase");
+
+    // getPastStoredBalance should match historical balances.
+    assertEq(
+      aToken.getPastStoredBalance(_who, block.number - _blocksJumped - _blocksJumpedSecondTime + 1),
+      _rawBalances[0],
+      "getPastStoredBalance did not match original raw balance"
+    );
+    assertEq(
+      aToken.getPastStoredBalance(_who, block.number - _blocksJumpedSecondTime + 1),
+      _rawBalances[1],
+      "getPastStoredBalance did not match raw balance after second supply"
+    );
+    // getPastStoredBalance should be able to give us the raw balance at intermediate points.
+    assertEq(
+      aToken.getPastStoredBalance(_who, block.number - _blocksJumpedSecondTime / 3), // random point
+      _rawBalances[1]
+    );
+    assertEq(
+      aToken.getPastStoredBalance(_who, block.number - _blocksJumpedSecondTime / 3),
+      aToken.getPastStoredBalance(_who, block.number - 1)
+    );
+
+    // Withdrawals should be reflected in getPastStoredBalance.
+    vm.startPrank(_who);
+    pool.withdraw(
+      address(govToken),
+      aToken.balanceOf(_who) / 3, // Withdraw 1/3rd of balance.
+      _who
+    );
+    vm.stopPrank();
+
+    // Advance the clock
+    uint256 _blocksJumpedThirdTime = 10;
+    vm.roll(block.number + _blocksJumpedThirdTime);
+    vm.warp(block.timestamp + 10 days);
+
+    assertEq(
+      aToken.getPastStoredBalance(_who, block.number - _blocksJumpedThirdTime),
+      aToken.exposedRawBalanceOf(_who)
+    );
+    assertEq(aToken.getPastStoredBalance(_who, block.number - 1), aToken.exposedRawBalanceOf(_who));
+    assertGt(
+      _rawBalances[1], // The raw balance pre-withdrawal.
+      aToken.getPastStoredBalance(_who, block.number - _blocksJumpedThirdTime)
+    );
+  }
+
+  function test_GetPastStoredBalanceHandlesTransfers() public {
+    _initiateRebasing();
+
+    address _userA = address(0xBABE);
+    address _userB = address(0xBEEF);
+    uint256 _amount = 4242 ether;
+
+    // Deposit.
+    _mintGovAndSupplyToAave(_userA, _amount);
+    uint256 _initRawBalanceUserA = aToken.exposedRawBalanceOf(_userA);
+    uint256 _initRawBalanceUserB = aToken.exposedRawBalanceOf(_userB);
+
+    // Advance the clock so that we checkpoint and let some rebasing happen.
+    vm.roll(block.number + 100);
+    vm.warp(block.timestamp + 100 days);
+
+    // Get the rebased balance for userA.
+    uint256 _initBalanceUserA = aToken.balanceOf(_userA);
+    uint256 _initBalanceUserB = aToken.balanceOf(_userB);
+    assertEq(_initBalanceUserB, 0);
+
+    // Transfer aTokens to userB.
+    vm.prank(_userA);
+    aToken.transfer(_userB, _initBalanceUserA / 3);
```

Comment:
Per above, should also have an `aToken.transferFrom` test

Reply:
Added in a324d17